packages/astro/src/assets/image-endpoint.ts

```diff
@@ -0,0 +1,65 @@
+import mime from 'mime';
+import type { APIRoute } from '../@types/astro.js';
+import { getConfiguredImageService, isRemoteImage } from './internal.js';
+import { isLocalService } from './services/service.js';
+import { etag } from './utils/etag.js';
+
+async function loadRemoteImage(src: URL) {
+	try {
+		const res = await fetch(src);
+
+		if (!res.ok) {
+			return undefined;
+		}
+
+		return Buffer.from(await res.arrayBuffer());
+	} catch (err: unknown) {
+		return undefined;
+	}
+}
+
+/**
+ * Endpoint used in SSR to serve optimized images
+ */
+export const get: APIRoute = async ({ request }) => {
+	try {
+		const imageService = await getConfiguredImageService();
+
+		if (!isLocalService(imageService)) {
+			throw new Error('Configured image service is not a local service');
+		}
+
+		const url = new URL(request.url);
+		const transform = await imageService.parseURL(url);
+
+		if (!transform || !transform.src) {
+			throw new Error('Incorrect transform returned by `parseURL`');
```

Comment:
We can definitely add more information to the message, however since this error is shown to users of the user's website, we actually need to be careful about what we say.